front/migrations/20241211_parents_migrator.ts

```diff
@@ -190,7 +190,128 @@ const migrators: Record<ConnectorProvider, ProviderMigrator | null> = {
   },
   google_drive: null,
   microsoft: null,
-  github: null,
+  github: {
+    transformer: (nodeId, parents) => {
+      if (nodeId !== parents[0]) {
+        logger.warn(
+          { nodeId, parents, problem: "Not enough parents" },
+          "Github nodeId !== parents[0]"
+        );
+        return { parents, parentId: parents[1] };
+      }
+      const repoId = parents[parents.length - 1];
+      // case where we are already good
+      if (repoId.startsWith("github-repository")) {
+        return { parents, parentId: parents[1] };
+      }
+      if (parents.length < 3) {
+        logger.warn(
+          { nodeId, parents, problem: "Not enough parents" },
+          "Invalid Github parents"
+        );
+        return { parents, parentId: parents[1] };
+      }
+      assert(/^\d+$/.test(repoId), `Invalid repoId: ${repoId}`);
+
+      if (nodeId.startsWith("github-issue-")) {
+        return {
+          parents: [
+            // old parents
+            ...parents,
+            // new parents
+            `github-issues-${repoId}`,
+            `github-repository-${repoId}`,
+          ],
+          parentId: parents[1],
+        };
+      }
+      if (nodeId.startsWith("github-discussion-")) {
+        return {
+          parents: [
+            // old parents
+            ...parents,
+            // new parents
+            `github-discussions-${repoId}`,
+            `github-repository-${repoId}`,
+          ],
+          parentId: parents[1],
+        };
+      }
+      if (nodeId.startsWith("github-code")) {
+        // github-code-${repoId}-file-xxx, github-code-${repoId}-dir-xxx, ..., github-code-${repoId}, ${repoId}
+        return {
+          parents: [
+            ...parents,
+            ...parents.slice(1, -2).reverse(),
+            `github-code-${repoId}`,
+            `github-repository-${repoId}`,
+          ],
+          parentId: parents[1],
+        };
+      }
+      throw new Error(`Unrecognized node type: ${nodeId}`);
+    },
+    cleaner: (nodeId, parents) => {
+      if (nodeId !== parents[0]) {
+        logger.warn(
+          { nodeId, parents, problem: "Not enough parents" },
+          "Github nodeId !== parents[0]"
+        );
+        return { parents, parentId: parents[1] };
+      }
+      const lastParent = parents[parents.length - 1];
+      // case where we are already good
+      if (!lastParent.startsWith("github-repository")) {
+        logger.warn(
+          { nodeId, parents, problem: "Not enough parents" },
+          "Github parents[-1] !== github-repository-xxx"
+        );
+        return { parents, parentId: parents[1] };
+      }
+      const repoId = lastParent.replace("github-repository-", "");
+      assert(/^\d+$/.test(repoId), `Invalid repoId: ${repoId}`);
+
+      if (nodeId.startsWith("github-code")) {
+        const parentsDir = parents.filter((p) => p.includes("-dir-"));
+
+        assert(
+          parentsDir.length % 2 === 0,
```

Comment:
actually, for the code nodes, wouldn't that be simpler to just rewrite the code dir parents directly in the right order in the transform, and not have to revert after?
wdyt?

Reply:
let's do that 👍

Reply:
done